### PR TITLE
feat(DLI): add a new resource to manage SQL template

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dli' TESTARGS='-run=TestAccSQLTemplate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run=TestAccSQLTemplate_basic -timeout 360m -parallel 4
=== RUN   TestAccSQLTemplate_basic
=== PAUSE TestAccSQLTemplate_basic
=== CONT  TestAccSQLTemplate_basic
--- PASS: TestAccSQLTemplate_basic (12.51s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli     12.554s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_template_sql_test.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+`, name)
+}
+
+func testSQLTemplate_basic_update(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_dli_template_sql" "test" {
+  name        = "%s"
+  sql         = "select * from t2"
+  group       = "demo_2"
+  description = "This is a updated demo"
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_template_sql_test.go]: ditto

File: huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_template_sql_test.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+				),
+			},
+			{
+				Config: testSQLTemplate_basic_update(name),
+				Check: resource.ComposeTestCheckFunc(
+					rc.CheckResourceExists(),
+					resource.TestCheckResourceAttr(rName, "name", name),
+					resource.TestCheckResourceAttr(rName, "sql", "select * from t2"),
+					resource.TestCheckResourceAttr(rName, "group", "demo_2"),
+					resource.TestCheckResourceAttr(rName, "description", "This is a updated demo"),
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_template_sql_test.go]:
```suggestion
					resource.TestCheckResourceAttr(rName, "description", "This is an updated demo"),
```

File: huaweicloud/services/dli/resource_huaweicloud_dli_template_sql.go [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+	deleteSQLTemplateOpt.JSONBody = utils.RemoveNil(buildDeleteSQLTemplateBodyParams(d, cfg))
+	_, err = deleteSQLTemplateClient.Request("POST", deleteSQLTemplatePath, &deleteSQLTemplateOpt)
+	if err != nil {
+		return diag.Errorf("error deleting SQLTemplate: %s", err)
+	}
+
+	return nil
+}
+
+func buildDeleteSQLTemplateBodyParams(d *schema.ResourceData, _ *config.Config) map[string]interface{} {
```
Comment (houpeng80) [inline huaweicloud/services/dli/resource_huaweicloud_dli_template_sql.go]: it is better to remove *config.Config

File: huaweicloud/services/dli/resource_huaweicloud_dli_template_sql.go [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+		updateSQLTemplateOpt.JSONBody = utils.RemoveNil(buildUpdateSQLTemplateBodyParams(d, cfg))
+		_, err = updateSQLTemplateClient.Request("PUT", updateSQLTemplatePath, &updateSQLTemplateOpt)
+		if err != nil {
+			return diag.Errorf("error updating SQLTemplate: %s", err)
+		}
+	}
+	return resourceSQLTemplateRead(ctx, d, meta)
+}
+
+func buildUpdateSQLTemplateBodyParams(d *schema.ResourceData, _ *config.Config) map[string]interface{} {
```
Comment (houpeng80) [inline huaweicloud/services/dli/resource_huaweicloud_dli_template_sql.go]: this func is same to buildCreateSQLTemplateBodyParams, they can be merged to one

File: huaweicloud/services/dli/resource_huaweicloud_dli_template_sql.go [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+	id, err := jmespath.Search("sql_id", createSQLTemplateRespBody)
+	if err != nil {
+		return diag.Errorf("error creating SQLTemplate: ID is not found in API response")
+	}
+	d.SetId(id.(string))
+
+	return resourceSQLTemplateRead(ctx, d, meta)
+}
+
+func buildCreateSQLTemplateBodyParams(d *schema.ResourceData, _ *config.Config) map[string]interface{} {
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/dli/resource_huaweicloud_dli_template_sql.go]: this func is same to buildUpdateSQLTemplateBodyParams, please them into one func

File: huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_template_sql_test.go [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+  group       = "demo"
+  description ="This is a demo"
+}
+`, name)
+}
+
+func testSQLTemplate_basic_update(name string) string {
+	return fmt.Sprintf(`
+resource "huaweicloud_dli_template_sql" "test" {
+  name        = "%s"
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_template_sql_test.go:122]: updating name shoud also be checked

File: docs/resources/dli_template_sql.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID.
+
+* `owner` - The user ID of owner.
+
+## Import
+
+The sql template can be imported using the `id`, e.g.
```
Comment (Jason-Zhang9309) [inline docs/resources/dli_template_sql.md]:
```suggestion
The SQL template can be imported using the `id`, e.g.
```

File: docs/resources/dli_template_sql.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    name        = "demo"
+    sql         = var.sql
+    group       = "test"
+    description = "This is a demo"
+  }
+```
+
+## Argument Reference
+
+the SQL template
```
Comment (Jason-Zhang9309) [inline docs/resources/dli_template_sql.md]: remove this line

Comment (hmbig2) [reply docs/resources/dli_template_sql.md]: done

File: docs/resources/dli_template_sql.md (hunk 7 above)
Comment (hmbig2) [reply docs/resources/dli_template_sql.md]: done

File: huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_template_sql_test.go (hunk 2 above)
Comment (hmbig2) [reply huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_template_sql_test.go]: done

File: huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_template_sql_test.go (hunk 6 above)
Comment (hmbig2) [reply huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_template_sql_test.go:122]: done

**Pull request merged**